src/file_cache.rs

```diff
@@ -10,43 +10,46 @@ use crate::ErrorExt as _;
 use crate::Result;
 
 
-#[derive(Debug)]
-struct Entry<T> {
+#[derive(Debug, Eq, Hash, PartialEq)]
+struct EntryMeta {
+    path: PathBuf,
     dev: libc::dev_t,
     inode: libc::ino_t,
     size: libc::off_t,
     mtime_sec: libc::time_t,
     mtime_nsec: i64,
-    file: File,
-    value: Option<T>,
 }
 
-impl<T> Entry<T> {
-    fn new(stat: &libc::stat, file: File) -> Self {
+impl EntryMeta {
+    fn new(path: PathBuf, stat: &libc::stat) -> Self {
         Self {
+            path,
             dev: stat.st_dev,
             inode: stat.st_ino,
             size: stat.st_size,
             mtime_sec: stat.st_mtime,
             mtime_nsec: stat.st_mtime_nsec,
```

Comment:
nit for a follow up: we are wasting a bit of memory here, we can just converting mtime to nsec and use one u64 field instead of having two separate fields.